cpp/ql/src/semmle/code/cpp/Type.qll

```diff
@@ -1276,7 +1276,7 @@ private predicate suppressUnusedThis(Type t) { any() }
 
 /** A source code location referring to a type */
 class TypeMention extends Locatable, @type_mention {
-  override string toString() {result = "mention of " + getMentionedType()}
+  override string toString() {result = "type mention"}
```

Comment:
Would it be sufficient to change this to `result = "mention of " + getMentionedType().getName()` to avoid the recursion? That would hopefully keep the more informative to-string.

Reply:
Yes, that would turn this into a PR that we can merge right away, without changes to tests or change note. I prefer `result = "type mention"` because then we avoid the time and space overhead of creating a new string for each type.

Another option would be `result = getMentionedType().getName()`, which would make the string informative again. It would be in line with variable accesses, whose `toString` is just the name of the variable. But as far as I know, nothing benefits from `TypeMention.toString` being informative -- except for one test that's easily fixed.